web/app/metrics-adapters/google-analytics-four.ts

```diff
@@ -0,0 +1,39 @@
+import { inject as service } from "@ember/service";
+import GoogleAnalyticsFour from "ember-metrics/metrics-adapters/google-analytics-four";
+import ConfigService from "hermes/services/config";
+
+declare global {
+  interface Window {
+    dataLayer: any[];
+  }
+}
+
+export default class GoogleAnalyticsFourAdapter extends GoogleAnalyticsFour {
+  @service("config") declare configSvc: ConfigService;
+
+  /**
+   * Modified from its parent class:
+   * https://github.com/adopted-ember-addons/ember-metrics/blob/master/addon/metrics-adapters/google-analytics-four.js#L11-L28
```

Comment:
Nit: might want to put the commit ID here in case future changes to this file in this branch change the line numbers.

```suggestion
   * https://github.com/adopted-ember-addons/ember-metrics/blob/01bb65e68f565fc48c2fa252d17a1b7d3d099e68/addon/metrics-adapters/google-analytics-four.js#L11-L28
```